docs/en/query_language/grant.md

```diff
@@ -0,0 +1,326 @@
+
+### ALTER {#grant-alter}
+
+Allows to perform [ALTER](alter.md) queries corresponding to the following hierarchy of privileges:
+
+- `ALTER`
+  - `ALTER TABLE`
+    - `UPDATE`
+    - `DELETE`
+    - `ALTER COLUMN`
+      - `ADD COLUMN`
+      - `DROP COLUMN`
+      - `MODIFY COLUMN`
+      - `COMMENT COLUMN`
+    - `INDEX`
+      - `ALTER ORDER BY`
+      - `ADD INDEX`
+      - `DROP INDEX`
+      - `MATERIALIZE INDEX`
+      - `CLEAR INDEX`
+    - `ALTER CONSTRAINT`
+      - `ADD CONSTRAINT`
+      - `DROP CONSTRAINT`
+    - `MODIFY TTL`
+    - `MODIFY SETTING`
+    - `MOVE PARTITION`
+    - `FETCH PARTITION`
+    - `FREEZE PARTITION`
+    - `ALTER VIEW`
+      - `REFRESH VIEW`
+      - `MODIFY VIEW QUERY`
+
+Examples of how this hierarchy is treated:
+
+- The `ALTER` privilege includes all other privileges. 
+- `ALTER CONSTRAINT` includes `ADD CONSTRAINT` and `DROP CONSTRAINT` privileges.
+
+**Notes**
+
+- The `MODIFY SETTING` privilege allows to modify table engine settings. In doesn't affect settings or server configuration parameters.
+- The `ATTACH` operation needs the [CREATE](#grant-create) privilege.
+- The `DETACH` operation needs the [DROP](#grant-drop) privilege.
+
+### CREATE {#grant-create}
+
+Allows to perform [CREATE](create.md) DDL-queries corresponding to the following hierarchy of privileges:
+
+- `CREATE`
+    - `CREATE DATABASE`
+    - `CREATE TABLE`
+      - `CREATE VIEW`
+    - `CREATE TEMPORARY TABLE`
+    - `CREATE DICTIONARY`
+
+**Notes**
+
+- The `CREATE` privilege doesn't allow a grantee to delete the created table. A user needs [DROP](#grant-drop).
+
+### DROP {#grant-drop}
+
+Allows to perform [DROP](misc.md#drop-statement) queries corresponding to the following hierarchy of privileges:
+
+- `DROP`
+  - `DROP DATABASE`
+  - `DROP TABLE`
+    - `DROP VIEW`
+  - `DROP DICTIONARY`
+
+### TRUNCATE {#grant-truncate}
+
+Allows to perform [TRUNCATE](misc.md#truncate-statement) queries corresponding to the following hierarchy of privileges:
+
+- `TRUNCATE`
+  - `TRUNCATE TABLE`
+    - `TRUNCATE VIEW`
+
+### OPTIMIZE {#grant-optimize}
+
+Allows to perform the [OPTIMIZE TABLE](misc.md#misc_operations-optimize) queries.
+
+
+### SHOW {#grant-show}
+
+Allows to perform the [SHOW CREATE TABLE](show.md#show-create-table-statement) queries.
+
+**Notes**
+
+A user has the `SHOW` privilege if it has any another right concerning the specified table.
+
+### EXISTS {#grant-exists}
+
+Allows to perform the [EXISTS](misc.md#exists-statement) queries.
+
+### KILL {#grant-kill}
+
+Allows to perform the [KILL](misc.md#kill-query-statement) queries corresponding to the following hierarchy of privileges:
+
+- `KILL`
+  - `KILL QUERY`
+  - `KILL MUTATION`
```

Comment:
The access type `KILL MUTATION` was removed in RBAC-7. To kill mutation you should have the same rights as for starting it, i.e. to execute the corresponding ALTER.